packages/core/core/generate-text/generate-text.ts

```diff
@@ -106,79 +110,165 @@ case of misconfigured tools.
 By default, it's set to 0, which will disable the feature.
      */
     maxToolRoundtrips?: number;
-  }): Promise<GenerateTextResult<TOOLS>> {
-  const retry = retryWithExponentialBackoff({ maxRetries });
-  const validatedPrompt = getValidatedPrompt({ system, prompt, messages });
 
-  const mode = {
-    type: 'regular' as const,
-    ...prepareToolsAndToolChoice({ tools, toolChoice }),
+    /**
+     * Optional telemetry information.
+     */
+    // This is meant to be both flexible for custom app requirements (metadata)
+    // and extensible for standardization (example: functionId, more to come).
+    telemetry?: {
+      functionId?: string;
+      metadata?: Record<string, AttributeValue>;
+    };
+  }): Promise<GenerateTextResult<TOOLS>> {
+  const baseTelemetryAttributes = {
+    'ai.model.provider': model.provider,
+    'ai.model.id': model.modelId,
+    // settings:
+    'ai.settings.maxRetries': maxRetries,
+    ...Object.entries(settings ?? {}).reduce((attributes, [key, value]) => {
+      attributes[`ai.settings.${key}`] = value;
+      return attributes;
+    }, {} as Record<string, AttributeValue>),
+    // special telemetry information
+    'ai.telemetry.functionId': telemetry?.functionId,
```

Comment:
Ideally we'd also add `resource.name` and `operation.name`. Some OTEL providers, especially DataDog, are very good at using these. One idea I have here:

```
'operation.name': 'ai.generateText'
'resource.name': telemetry?.functionId ?? 'something_something'
```